[CB-373] Confirm 팝업 리팩토링

dmstmdrbs opened the pull request:
[CB-373]

### 🔨 Jira 태스크

- [CB-373] Confirm 팝업 리팩토링

### 📐 구현한 내용

-  새로운 Confirm 팝업 컴포넌트 추가
   - 기존의 ToastConfirm을 대체하기 위한 컴포넌트
   - useConfirm 훅을 사용하여 리턴값으로 받아온 배열의 첫 번째 원소인 함수를 이용하여 팝업을 open
      - promise를 반환하며, 팝업 확인 시 true, 취소 시 false 반환
	  - asyc/await를 꼭 사용해야 함
	  - 팝업의 내용을 openPopup의 인자로 넘겨주어야 함
- ToastConfirm 팝업에서 새로운 팝업으로 변경
  - 기존의 ToastConfirm 사용 부분을 새로운 팝업으로 대체
  - ToastConfirm 팝업 삭제
  
### 🚧 논의 사항

-


[CB-373]: https://cocobob.atlassian.net/browse/CB-373?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CB-373]: https://cocobob.atlassian.net/browse/CB-373?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

dmstmdrbs committed:
CB-373 feat: add new Popup components with useConfirm hooks

새로운 Confirm 팝업 컴포넌트 추가
- 기존의 ToastConfirm을 대체하기 위한 컴포넌트
- useConfirm 훅을 사용하여 리턴값으로 받아온 배열의 첫 번째 원소인 함수를 이용하여 팝업을 open
  - promise를 반환하며, 팝업 확인 시 true, 취소 시 false 반환
	- asyc/await를 꼭 사용해야 함
	- 팝업의 내용을 openPopup의 인자로 넘겨주어야 함

dmstmdrbs committed:
CB-373 refactor: change popup from ToastConfirm to new popup

ToastConfirm 팝업에서 새로운 팝업으로 변경
- 기존의 ToastConfirm 사용 부분을 새로운 팝업으로 대체
- ToastConfirm 팝업 삭제

dmstmdrbs committed:
CB-373 refactor: add useConfirm file

useConfirm 훅을 다른 파일로 분리

taehong0-0 commented on src/components/Confirm/ConfirmModal.tsx:
```diff
@@ -0,0 +1,71 @@
+  const popupOpened = useAppSelector(isPopupOpened);
+  const { title, contents, promiseInfo } = useContext(ConfirmContext);
+
+  const closePopup = () => dispatch(setPopupOpened(false));
```
혹시 위의 hidePopup과 같은 역할을 하는 것 처럼 보이는데 따로 선언하신 이유가 있는건가요?

dmstmdrbs replied:
useConfirm은 외부에서 ConfirmPopup를 다루기 위해서 선언해두었습니다.
ConfirmModal 컴포넌트에서는 useConfirm 훅을 사용하는 것이 용도에 맞지 않다고 생각해서 이 부분에서는 새롭게 선언 해였습니다!

taehong0-0 reviewed (approved):
고생하셨습니다 !

dmstmdrbs committed:
CB-375 refactor: modify module import/export

모듈 import/export 관리 방식 변경
- index에서 취합하여 export 시킴.
- 한 곳에서 하위 폴더의 컴포넌트를 모두 가져올 수 있음

dmstmdrbs committed:
Merge pull request #71 from PinkDumbbell/CB-375-refactor/module

[CB-375] 모듈 관리 변경
